p2p/protocol/adv/adv.go

```diff
@@ -0,0 +1,122 @@
+package adv
```

Comment:
I'm not convinced by this package name.

maybe the protocol would make sense as 'head'? that's what this endpoint is called in the HTTP implementation - may be nice to have the two harmonized.

Reply:
agreed about name, I actually meant to bring that up in the PR description.

`head` seems fine. It's pretty specific, but this thing only does that, so works for me.